### PR TITLE
Properly detect root of a tree in Operation.SearchParentOperation

### LINKED ISSUE #22964: IOperation.Parent throws System.NullReferenceException on a global statement
```
        [CompilerTrait(CompilerFeature.IOperation)]
        [Fact]
        public void GlobalStatement_Parent()
        {
            var source =
@"
System.Console.WriteLine();
";
            var compilation = CreateStandardCompilation(source, options: TestOptions.ReleaseExe.WithScriptClassName("Script"), parseOptions: TestOptions.Script);
            compilation.VerifyDiagnostics();

            var tree = compilation.SyntaxTrees.Single();
            var statement = tree.GetRoot().DescendantNodes().OfType<StatementSyntax>().Single();
            var model = compilation.GetSemanticModel(tree);
            var operation = model.GetOperationInternal(statement);

            Assert.Equal(OperationKind.ExpressionStatement, operation.Kind);
            Assert.Null(operation.Parent);
        }
```

System.NullReferenceException: 'Object reference not set to an instance of an object.'

```
>	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MemberSemanticModel.GetBindingRoot(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node) Line 1336	C#	Symbols loaded.
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MemberSemanticModel.GetBindingRootOrInitializer(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node) Line 979	C#	Symbols loaded.
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MemberSemanticModel.GetOperationWorker(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node, System.Threading.CancellationToken cancellationToken) Line 964	C#	Symbols loaded.
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.GetOperationCore(Microsoft.CodeAnalysis.SyntaxNode node, System.Threading.CancellationToken cancellationToken) Line 469	C#	Symbols loaded.
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.SemanticModel.GetOperationInternal(Microsoft.CodeAnalysis.SyntaxNode node, System.Threading.CancellationToken cancellationToken) Line 86	C#	Symbols loaded.
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Operation.SearchParentOperation() Line 278	C#	Symbols loaded.
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Operation.Parent.get() Line 47	C#	Symbols loaded.
 	Roslyn.Compilers.CSharp.Semantic.UnitTests.dll!Microsoft.CodeAnalysis.CSharp.UnitTests.IOperationTests.GlobalStatement_Parent() Line 146	C#	Symbols loaded.
```

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #22964
vso :https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20Productivity/_workitems/edit/518796

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler, @dotnet/analyzer-ioperation, @heejaechang, @cston Please review 15.5 IOperation fix.

**Review by genlu — APPROVED** [review]
:shipit:

**Review by heejaechang — APPROVED** [review]

**Review by mavasani — APPROVED** [review]

File: src/Compilers/Core/Portable/Operations/Operation.cs [hunk 1]
```diff
@@ -286,6 +285,12 @@ internal IOperation SearchParentOperation()
                         }
                     }
 
+                    if (SemanticModel.Root == currentCandidate)
```
Comment (agocke) [inline src/Compilers/Core/Portable/Operations/Operation.cs]: I believe you could avoid duplicating this check by doing it at the beginning of the loop and setting `currentCandidate` to `Syntax` at the beginning of the loop.

Comment (AlekseyTs) [reply src/Compilers/Core/Portable/Operations/Operation.cs]: > I believe you could avoid duplicating this check by doing it at the beginning of the loop and setting currentCandidate to Syntax at the beginning of the loop.

Then the loop would have to be changed to ```while (true) and the null check moved inside. I believe the condition is simple enough that duplication is not an issue.

**Review by jcouv — APPROVED** [review]
LGTM

**Review by agocke — APPROVED** [review]
LGTM

**Comment by jinujoseph** [comment]
Pre-Approved to merge via [Link](https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20Productivity/_workitems/edit/518796)

**Pull request merged**